tidy3d/components/eme/simulation.py

```diff
@@ -0,0 +1,335 @@
+            The z range if plotting on xz or yz planes, y plane if plotting on xy plane.
+
+        Returns
+        -------
+        matplotlib.axes._subplots.Axes
+            The supplied or created matplotlib axes.
+        """
+
+        hlim, vlim = Scene._get_plot_lims(
+            bounds=self.simulation_bounds, x=x, y=y, z=z, hlim=hlim, vlim=vlim
+        )
+
+        ax = self.scene.plot_structures(ax=ax, x=x, y=y, z=z, hlim=hlim, vlim=vlim)
+        ax = self.plot_sources(ax=ax, x=x, y=y, z=z, hlim=hlim, vlim=vlim, alpha=source_alpha)
+        ax = self.plot_monitors(ax=ax, x=x, y=y, z=z, hlim=hlim, vlim=vlim, alpha=monitor_alpha)
+        ax = Scene._set_plot_bounds(
+            bounds=self.simulation_bounds, ax=ax, x=x, y=y, z=z, hlim=hlim, vlim=vlim
+        )
+        ax = self.plot_boundaries(ax=ax, x=x, y=y, z=z)
+        ax = self.plot_symmetries(ax=ax, x=x, y=y, z=z, hlim=hlim, vlim=vlim)
+
+        ax = self.plot_eme_grid(ax=ax, x=x, y=y, z=z, hlim=hlim, vlim=vlim)
+
+        return ax
+
+    @cached_property
+    def eme_grid(self) -> EMEGrid:
+        """The EME grid as defined by 'eme_grid_spec'.
+        An EME grid is a 1D grid aligned with the propagation axis,
+        dividing the simulation into cells. Modes and mode coefficients
+        are defined at the central plane of each cell. Typically,
+        cell boundaries are aligned with interfaces between structures
+        in the simulation.
+
+        This is distinct from 'grid', which is the grid used in the tangential directions
+        as well as the grid used for field monitors.
+        """
+        return self.eme_grid_spec.make_grid(center=self.center, size=self.size, axis=self.axis)
+
+    @classmethod
+    def from_scene(cls, scene: Scene, **kwargs) -> EMESimulation:
+        """Create an EME simulation from a :class:.`Scene` instance. Must provide additional parameters
+        to define a valid EME simulation (for example, ``size``, ``grid_spec``, etc).
+
+        Parameters
+        ----------
+        scene : :class:.`Scene`
+            Scene containing structures information.
+        **kwargs
+            Other arguments
+
+        """
+        return cls(
+            structures=scene.structures,
+            medium=scene.medium,
+            **kwargs,
+        )
+
+    def _to_fdtd_sim(self, **kwargs) -> Simulation:
+        """Construct an FDTD simulation from an EME simulation. Must provide additional
+        parameters to define a valid FDTD simulation (for example, ``run_time``).
+
+        Used to generate the FDTD grid.
+        """
+        return Simulation(
+            center=self.center,
+            size=self.size,
+            medium=self.medium,
+            structures=self.structures,
+            symmetry=self.symmetry,
+            grid_spec=self.grid_spec,
+            boundary_spec=self.boundary_spec,
+            version=self.version,
+            **kwargs,
+        )
+
+    @cached_property
+    def grid(self) -> Grid:
```

Comment:
Currently, the grid is generated by the grid_spec using the ordinary grid generation code, by converting the sim to an fdtd sim first. This is probably the best way if you just care about the EME simulation. However, it is possible that you have an FDTD simulation with a pre-existing grid, and you want to make sure the EME simulation is using exactly the same grid. In this case, the grid generation here won't generate exactly the same grid. So I might add another field for if the user wants to specify an actual grid to override the grid_spec.

Reply:
Hmm this is also pretty interesting in the context of grid translations between other solvers, I wonder to what level do we want other solvers to have this functionality too

Reply:
btw, now the grid generation is handled in AbstractYeeGridSimulation instead of via an intermediate fdtd simulation.